lib/utils.js

```diff
@@ -575,9 +575,15 @@ exports.lookupFiles = function lookupFiles(filepath, extensions, recursive) {
   var stat;
 
   if (!fs.existsSync(filepath)) {
-    if (fs.existsSync(filepath + '.js')) {
-      filepath += '.js';
-    } else {
+    // check all extensions
+    if (
+      !extensions.some(function(ext) {
+        if (fs.existsSync(filepath + '.' + ext)) {
+          filepath += '.' + ext;
+          return true;
+        }
+      })
+    ) {
```

Comment:
The _only_ reason this was here was to allow the historical usage of "test.js" via the default spec "test". IMO, we should **not** propagate this further.

The "Handle directory" code below already handles extensions for "test/*".